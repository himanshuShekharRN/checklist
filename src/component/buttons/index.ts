import {FloatingAddButton} from './floatingAddButton';
import {RoundedButton} from './roundedButton';

export {FloatingAddButton, RoundedButton};
