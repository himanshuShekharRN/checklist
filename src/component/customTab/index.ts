import {CustomTab} from './CustomTab';
import {FirstRoute} from './firstRoute';
import {SecondRoute} from './secondRoute';
import {ThirdRoute} from './thirdRoute';
import {FourthRoute} from './fourthRoute';
import {FifthRoute} from './fifthRoute';

export {
  CustomTab,
  FirstRoute,
  SecondRoute,
  ThirdRoute,
  FourthRoute,
  FifthRoute,
};
