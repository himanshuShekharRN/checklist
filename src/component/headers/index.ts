import {HeadersWithButton} from './headersWithButton/HeadersWithButton';

export {HeadersWithButton};
