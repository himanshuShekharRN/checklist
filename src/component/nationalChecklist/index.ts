import {NationalChecklist} from './NationalChecklist';

export {NationalChecklist};
