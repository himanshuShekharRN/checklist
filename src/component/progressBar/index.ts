import {LinearProgressBar} from './linearProgressBar';

export {LinearProgressBar};
