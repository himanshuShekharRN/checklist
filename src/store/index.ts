import createSagaMiddleware from 'redux-saga';
import {configureStore} from '@reduxjs/toolkit';
import {persistReducer, persistStore} from 'redux-persist';
import AsyncStorage from '@react-native-async-storage/async-storage';

import {appReducer} from './reducer';
import {rootSaga} from '../network/saga';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: ['checkListReducer', 'departureCheckListReducer'],
};

const sagaMiddleware = createSagaMiddleware();
const persistedReducer = persistReducer(persistConfig, appReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: [sagaMiddleware],
});

export const persistor = persistStore(store);
sagaMiddleware.run(rootSaga);
