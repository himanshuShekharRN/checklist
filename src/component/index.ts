import {SwipableCards} from './swipableCards/SwipableCards';
import {SwipableActionButton} from './swipableActionButton/SwipableActionButton';
import {Card} from './cards';
import {MyCheckList} from './myChecklist';
import {Space} from './space';
import {PreDepartureList} from './preDepartureList';
import {Title} from './title';
import {HeadersWithButton} from './headers';
import {CustomSVG} from './customSVG';
import {CustomIcon} from './customIcon';
import {FloatingAddButton, RoundedButton} from './buttons';
import {ListCard} from './listCard';
import {LinearProgressBar} from './progressBar';
import {CustomTab} from './customTab';
import {DepartureDocCard} from './departureDocCard';
import {NationalChecklist} from './nationalChecklist';

export {
  FloatingAddButton,
  Card,
  MyCheckList,
  Space,
  PreDepartureList,
  Title,
  HeadersWithButton,
  CustomSVG,
  CustomIcon,
  RoundedButton,
  ListCard,
  LinearProgressBar,
  CustomTab,
  DepartureDocCard,
  NationalChecklist,
  SwipableActionButton,
  SwipableCards,
};
