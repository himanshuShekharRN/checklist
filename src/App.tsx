import React from 'react';
import 'react-native-gesture-handler';

import {Navigation} from './navigation';
import {Provider} from 'react-redux';
import {persistor, store} from './store';
import {PersistGate} from 'redux-persist/integration/react';

export const App: React.FC = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Navigation />
      </PersistGate>
    </Provider>
  );
};
