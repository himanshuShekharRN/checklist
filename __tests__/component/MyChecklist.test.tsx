import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {MyCheckList} from '../../src/component';
import {Provider} from 'react-redux';
import {store} from '../../src/store';

describe('MyCheckList', () => {
  let tree: any = render(
    <Provider store={store}>
      <MyCheckList />
    </Provider>,
  );

  test('renders correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
  });

  test('renders title correctly', () => {
    render(
      <Provider store={store}>
        <MyCheckList />
      </Provider>,
    );

    const titleContent = screen.getByText('My Checklists');
    expect(titleContent).toBeTruthy();
  });

  test('renders sub title correctly', () => {
    render(
      <Provider store={store}>
        <MyCheckList />
      </Provider>,
    );

    const titleContent = screen.getByText('Create your own personal checklist');
    expect(titleContent).toBeTruthy();
  });

  test('snapshot created successfully', () => {
    render(
      <Provider store={store}>
        <MyCheckList />
      </Provider>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
