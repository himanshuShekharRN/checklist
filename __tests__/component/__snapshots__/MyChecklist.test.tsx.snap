// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCheckList snapshot created successfully 1`] = `
[
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "Roboto-Medium",
        "fontSize": 16,
        "textAlign": "left",
      }
    }
  >
    My Checklists
  </Text>,
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "Roboto-Italic",
        "fontSize": 12,
        "marginTop": 4,
        "textAlign": "left",
      }
    }
  >
    Create your own personal checklist
  </Text>,
  <View
    style={
      {
        "backgroundColor": undefined,
        "height": 12,
        "width": undefined,
      }
    }
  />,
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    bounces={false}
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginTop": 80,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000000",
              "fontFamily": "Roboto-Regular",
              "fontSize": 12,
              "textAlign": "left",
            }
          }
        >
          No Personal Checklist Found!
        </Text>
      </View>
    </View>
  </RCTScrollView>,
]
`;
